webpack/components/RolesIndex/RolesIndex.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import { Table, TableHeader, TableBody } from '@patternfly/react-table';
+import { Pagination } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { linkCellFormatter, DEFAULT_PER_PAGE } from './RolesIndexConstants';
+
+const RolesIndex = ({ rowsData, RolesIndexactions }) => {
+  const cols = [
+    { title: 'Name' },
+    { title: 'Variables', cellTransforms: [linkCellFormatter] },
+    { title: 'Hostgroups' },
+    { title: 'Hosts', cellTransforms: [linkCellFormatter] },
+    { title: 'Imported at' },
+  ];
+  const rows = rowsData;
+  const [page, setPage] = useState();
+  const [perPage, setPerPage] = useState(DEFAULT_PER_PAGE);
+  const [paginatedRows, setPaginatedRows] = useState(rows.slice(0, perPage));
+
+  const handlePerPageSelect = (
+    event,
+    newPerPage,
+    newPage,
+    startIdx,
+    endIdx
+  ) => {
+    setPerPage(newPerPage);
+    setPage(newPage);
+    setPaginatedRows(rows.slice(startIdx, endIdx));
+  };
+
+  const handleSetPage = (event, newPage) => {
```

Comment:
I think I saw a similar functions in [another place](https://github.com/theforeman/foreman_ansible/blob/master/webpack/components/AnsibleRolesAndVariables/AnsibleRolesAndVariables.js#L71). Would it be possible to reuse the common parts? This could be a case for a higher order component.

Same could be possibly done for `handlePerPageSelect`.

Reply:
yep, I will do it.